scripts/travis/publish-tag.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Ensure that this is being run by Travis
+if [ "$TRAVIS" != "true" ] || [ "$USER" != "travis" ]; then
+  echo "This script should only be run by Travis CI."
+  exit 2
+fi
+
+# Ensure that the tag is named properly as a semver tag
+if [[ ! "$TRAVIS_TAG" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo "Tag $TRAVIS_TAG is NOT a valid semver tag (vX.Y.Z), please delete this tag."
+  exit 1
+fi
+
+# Ensure that the script is being run from the root project directory
+PROPERTIES_FILE='gradle.properties'
+if [ ! -f "$PROPERTIES_FILE" ]; then
+  echo "Could not find $PROPERTIES_FILE, are you sure this is being run from the root project directory?"
+  echo "PWD: ${PWD}"
+  exit 1
+fi
+
+# Determine the version being published
+VERSION=$(awk 'BEGIN { FS = "=" }; $1 == "version" { print $2 }' $PROPERTIES_FILE | awk '{ print $1 }')
+if [ -z "$VERSION" ]; then
+  echo "Could not read the version from $PROPERTIES_FILE, please fix it and try again."
+  exit 1
+fi
+
+# Ensure the tag corresponds to the current version
+EXPECTED_TAG="v$VERSION"
+if [ "$TRAVIS_TAG" != "$EXPECTED_TAG" ]; then
+  echo "Attempting to publish ParSeq version $VERSION from tag $TRAVIS_TAG is illegal."
+  echo "Please delete this tag and publish instead from tag $EXPECTED_TAG"
+  exit 1
+fi
+
+# Ensure the commit environment variable exists
+if [ -z "$TRAVIS_COMMIT" ]; then
+  echo 'Cannot find environment variable named TRAVIS_COMMIT, did the Travis API change?'
+  exit 1
+fi
+
+# Ensure that the tag commit is an ancestor of master
+git merge-base --is-ancestor $TRAVIS_COMMIT master
```

Comment:
what about hotfix release? should those be done manually from local?
It might be good to add this script on any release tag or have a task for adding a hotfix script, so that we don't need to revert all changes after bad code.

Reply:
I think that case will have to be done manually on the command line. It'll be a rare case and I think having more stringent checks in the git tag deployment is more important than improving the dev experience of this rare corner case. Thanks for pointing this out though, this is something that will need to be documented in the updated dev guide.